packages/cozy-procedures/src/components/Amount.jsx

```diff
@@ -1,22 +1,74 @@
 import React from 'react'
 import PropTypes from 'prop-types'
+import { withRouter } from 'react-router'
+import get from 'lodash/get'
 import Topbar from './Topbar'
-import { translate } from 'cozy-ui/transpiled/react/I18n'
+import {
+  Title,
+  translate,
+  Label,
+  Input,
+  Button
+} from 'cozy-ui/transpiled/react/'
+import creditApplicationTemplate from '../templates/creditApplicationTemplate'
 
 class Amount extends React.Component {
+  constructor(props) {
+    super(props)
+    this.confirm = this.confirm.bind(this)
+    this.inputRef = React.createRef()
+  }
+
   render() {
-    const { t } = this.props
+    const { t, amount, updateAmount } = this.props
+    const min = get(creditApplicationTemplate, 'procedureData.amount.min')
+    const max = get(creditApplicationTemplate, 'procedureData.amount.max')
+    const defaultValue = get(
+      creditApplicationTemplate,
+      'procedureData.amount.default'
+    )
     return (
       <div>
         <Topbar title={t('amount.title')} />
-        Amount
+        <Title>{t('amount.subtitle')}</Title>
+
+        <div>
+          <Label htmlFor="amount-input">{t('amount.label')}</Label>
+          <Input
+            type="number"
+            id="amount-input"
+            min={min}
+            max={max}
+            value={amount || defaultValue}
+            onChange={e => updateAmount(parseInt(e.target.value))}
```

Comment:
maybe we should use `debounce` for all the inputs to avoid writing the redux state (and thus cause re-render) too often https://lodash.com/docs/4.17.11#debounce
(it is also the case for the personal data form)

Reply:
I just gave it a quick try, but it's not as easy as I thought. We still want the UI to be responsive, so we would need to store the user value in a state, use that in the UI, and only dispatch later… I don't think we need this extra complexity at the moment.